examples/erosion2/storage.py

```diff
@@ -1,33 +1,64 @@
 import shelve
+import string
+import time
+
+
+_CHARS = sorted(string.ascii_lowercase + string.digits)
+_N_CHARS = len(_CHARS)
+
+
+def _encode_id(num):
+    alias = ""
+    while num:
+        alias = _CHARS[num % _N_CHARS] + alias
+        num //= len(_CHARS)
+    return alias if alias else _CHARS[0]
 
 
 class LinkDB:
     def __init__(self, db_path):
         self.db_path = db_path
-
-    def add_link(self, alias=None, *, target_url, expiry_time, max_count):
-        entry = {
-            "target": target_url,
-            "alias": alias,
-            "expiry_time": expiry_time,
-            "max_count": max_count,
-            "count": 0,
-        }
         with shelve.open(self.db_path) as db:
-            db[alias] = entry
+            db["last_id"] = 41660
+            db["entries"] = {}
+
+    def add_link(self, target_url, alias=None, expiry_time=0, max_count=0):
+        with shelve.open(self.db_path, writeback=True) as db:
+            next_id = db["last_id"] + 1
+            if not alias:
+                alias = _encode_id(next_id)
+            if alias in db:
+                raise ValueError("alias already in use %r" % alias)
```

Comment:
It may be a little bit advanced for this application, but if the alias is in use but also autogenerated, we should probably generate the next one. On the other hand, if it was input by the user, it should result in a 400 of sorts, so they can pick a new alias.

Reply:
(as it is, an uncaught ValueError will be a 500/trigger the debugger)

Reply:
I'm aware of this one. I wanted to mimic the original erosion app here but I couldn't get it to work :( It required Py2 and then gave an error about strings. Anyway, I'll try to come up with something not-so-complicated.

Reply:
I think I've implemented an acceptable solution using a second cookie. Good thing this part is not in the text of the documentation.

Reply:
I'm marking this "resolved" but reopen if you disagree.